### ISSUE TITLE
Deleting daemon controller hangs

### ISSUE THREAD

**Issue by thockin** [kubernetes/kubernetes#14712]
I've left it alone for minutes now and kubectl is just hung.  I am not sure who owns DaemonSet right now...

Worse, I can delete the namespace, which completes successfully, but if I then recreate the same namespace the hung daemonset still exists!  Deleting a namespace should not complete if there are still objects alive!!

@derekwaynecarr

**Comment by dchen1107** [comment]
cc/ @mikedanese who currently own daemon controller.

**Comment by mikedanese** [comment]
I'll take a look at the reaper. #14666 fixes the namespace job deletion.

**Comment by thockin** [comment]
Seriously?  We have an open-coded delete for namespaces...sigh.  That
really needs to be auto-registered or something.

**Comment by derekwaynecarr** [comment]
@thockin - I agree that we need a way to auto-register.  It's on my TODO list to try and improve this, but I need some more features in our client libraries to do auto-discovery of registered API groups/resources.  See discussion: https://github.com/kubernetes/kubernetes/issues/14562

**Comment by erictune** [comment]
This could be #14682

**Comment by thockin** [comment]
I just did a dev-build-and-push and it still hangs :(

**Comment by thockin** [comment]
Nope.  It still hangs forever

**Comment by thockin** [comment]
```
apiVersion: experimental/v1alpha1
kind: DaemonSet
metadata:
  namespace: kube-demos
  name: daemons-demo-daemon
spec:
  template:
    metadata:
      labels:
        demo: daemons
    spec:
      nodeSelector:
        color: red
      containers:
      - name: hostname
        image: kubernetes/serve_hostname
```

**Comment by erictune** [comment]
@lavalamp, is use of fuzzer.Fuzz in pkg/kubectl/stop.go around line 201 to
generate a random node name considered a recommended use of fuzzer?

**Comment by mikedanese** [comment]
@erictune I have root caused this.

**Comment by lavalamp** [comment]
@erictune no, that is definitely increasing the WTFs-per-second of the codebase.

**Comment by erictune** [comment]
Was finding the root cause exacerbated by the fact that that in the
wait.Poll loop in
   func (reaper *DaemonSetReaper) Stop(..)  ignores errors?

**Comment by thockin** [comment]
zoinks!  Using fuzzer outside of a test should be verboten!

**Comment by mikedanese** [comment]
Fixed by #15196

**Comment by mikedanese** [comment]
@erictune it's been quite a journey.
